### PR TITLE
[16.0][FIX] account_invoice_supplier_self_invoice: use name over ref in reports

### PR DESCRIPTION
**Pull request by nicolas-delbovier-acsone**
The invoice 'ref' field is an external identifier provided by the vendor, not the primary internal identifier for the record. For consistency and proper identification, the 'name' field (Invoice Number) should always be displayed on the report regardless of the document type.

### CODE REVIEW

**Comment by nicolas-delbovier-acsone** [comment]
@jbaudoux

**Review by jbaudoux — CHANGES_REQUESTED** [review]
Better change xpath position="replace" by position="before" and then drop o.name

And also drop the section
```
 <xpath
            expr="//div[@id='informations']/div[@name='reference']"
            position="attributes"
        >
```

**Comment by jbaudoux** [comment]
cc @Shide

**Comment by Shide** [comment]
@nicolas-delbovier-acsone Can you provide a video with the process to know how it looks the report?

**Comment by Shide** [comment]
The report changes positions of partner and company.
The invoice looks like if has been generated by the partner, so the name that displays is the generated invoice number (self-billing number)
I need to test this in a runbot when comes up or you provide a video showing the pdf result

**Review by jbaudoux — APPROVED** [review]

**Comment by jbaudoux** [comment]
> The report changes positions of partner and company. The invoice looks like if has been generated by the partner, so the name that displays is the generated invoice number (self-billing number) I need to test this in a runbot when comes up or you provide a video showing the pdf result

@Shide This is being addressed here https://github.com/OCA/account-invoicing/pull/2254

File: account_invoice_supplier_self_invoice/views/report_self_invoice.xml [hunk 1]
```diff
@@ -129,16 +129,14 @@
...
             expr="//div[hasclass('page')]/h2/span[@t-field='o.name']"
-            position="replace"
+            position="before"
         >
             <span
                 t-if="o.set_self_invoice and o.move_type == 'in_refund'"
             >Self-bill Credit Note</span>
             <span
                 t-if="o.set_self_invoice and o.move_type == 'in_invoice'"
             >Self-bill Invoice</span>
```
Comment (jbaudoux) [inline account_invoice_supplier_self_invoice/views/report_self_invoice.xml:139]:
```suggestion
>Self-billed Credit Note</span>
<span
t-if="o.set_self_invoice and o.move_type == 'in_invoice'"
>Self-billed Invoice</span>
```

Comment (nicolas-delbovier-acsone) [reply account_invoice_supplier_self_invoice/views/report_self_invoice.xml:139]: I will make these changes here: https://github.com/OCA/account-invoicing/pull/2256

**Review by marcos-mendez — CHANGES_REQUESTED** [review]
**Automated Review -- Tests Failed**
### 1. **Root Cause**

The test failures occur because the `account.move` records are being created without required fields like `date` or `journal_id`, which are accessed during report rendering. This is likely due to incomplete test data setup in the `setUp` method, where the invoice is created without proper defaults for these fields.

### 2. **Suggested Fix**

In `/tmp/repo/account_invoice_supplier_self_invoice/tests/test_self_invoice.py`, the `setUp` method (line ~29) creates an `account.move` record without setting required fields such as `journal_id` and `date`. These fields must be explicitly provided or defaulted during creation.

**Fix:**
```python
self.invoice = self.env["account.move"].create({
    "move_type": "in_invoice",
    "partner_id": self.partner.id,
    "journal_id": self.env["account.journal"].search([("type", "=", "purchase")], limit=1).id,
    "date": fields.Date.today(),
    "set_self_invoice": True,
})
```

Ensure all test invoices are initialized with required fields to prevent `CacheMiss` errors during report rendering.

### 3. **Additional Code Issues**

- **Missing default journal in test setup**: The test does not ensure a valid `journal_id` is assigned, which causes a `CacheMiss` when the report tries to access `journal_id`.
- **Report template changes**: The removal of `o.ref` and `o.name` from the report template may affect display logic if not properly handled elsewhere; however, this is not a bug per se but should be validated in tests.

### 4. **Test Improvements**

Add tests using `SavepointCase` or `TransactionCase` to cover edge cases like:
- Invoice without `set_self_invoice`
- Invoice with `ref` and `name` fields
- Invoice with `journal_id` and `date` missing (to validate proper defaults or error handling)

Example test:
```python
def test_invoice_with_ref_and_name(self):
    invoice = self.env["account.move"].create({
        "move_type": "in_invoice",
        "partner_id": self.partner.id,
        "journal_id": self.journal.id,
        "date": fields.Date.today(),
        "ref": "TEST-REF",
        "name": "INV-001",
    })
    self.assertEqual(invoice.ref, "TEST-REF")
```

Use tagged tests for regression coverage:
```python
@tagged('post_install', 'manual_install')
def test_self_invoice_report_rendering(self):
    ...
```
---
_Environment via [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer): Minikube + K8s Job + oca-ci/py3.10-odoo16.0 | Odoo 16.0_
_Automated review by [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer) + qwen3-coder:30b_

**Review by BhaveshHeliconia — APPROVED** [review]
Code review LGTM!

**Comment by rousseldenis** [comment]
@nicolas-delbovier-acsone Could you rebase ?
